src/app/src/pages/redteam/setup/components/RunOptions.tsx

```diff
@@ -69,24 +110,45 @@ export const RunOptionsContent: React.FC<RunOptionsProps> = ({
             />
           )
         }
-        value={runOptions?.delay ?? 0}
+        value={delayInput}
         disabled={!canSetDelay}
         onChange={(e) => {
```

Comment:
Are the `onChange`, `onBlur`, and `onKeyDown` logics similar enough across the modified instances where a well-documented shared component is preferable?  If not, I'd like these to be better documented.

Reply:
They differ enough that I think adding a shared component would introduce unnecessary complexity. I've added comments above each input to document their behavior.